Dungeons/Dungeons/CorridorNodeLayouter.cs

```diff
@@ -94,204 +94,183 @@ public DungeonLevel DoLayout(List<DungeonNode> nodes, LayouterOptions opt = null
+      mazeNodes[6].GenerateLayoutDoorsCorrindor(EntranceSide.Top);
+      mazeNodes[7].GenerateLayoutDoorsCorrindor(EntranceSide.Left);
+      mazeNodes[7].GenerateLayoutDoorsCorrindor(EntranceSide.Right);
+      mazeNodes[8].GenerateLayoutDoorsCorrindor(EntranceSide.Bottom);
+      mazeNodes[8].GenerateLayoutDoorsCorrindor(EntranceSide.Top);
+
+      mazeNodes[9].GenerateLayoutDoorsCorrindor(EntranceSide.Top);
+      mazeNodes[9].GenerateLayoutDoorsCorrindor(EntranceSide.Bottom);
+      mazeNodes[10].GenerateLayoutDoorsCorrindor(EntranceSide.Top);
+      mazeNodes[10].GenerateLayoutDoorsCorrindor(EntranceSide.Bottom);
+      mazeNodes[11].GenerateLayoutDoorsCorrindor(EntranceSide.Left);
+      mazeNodes[11].GenerateLayoutDoorsCorrindor(EntranceSide.Right);
+      mazeNodes[12].GenerateLayoutDoorsCorrindor(EntranceSide.Right);
+      mazeNodes[12].GenerateLayoutDoorsCorrindor(EntranceSide.Left);
+
+      for (int i = 0; i<4;i++) //placing corrindors
       {
-        if (currentNode.NodeIndex == 0)
-          nextAppendInfo.side = GetRandSide();
-        else
-        {
-          if (info != null && info.ForcedNextRoomSide != null)
-            nextAppendInfo.side = info.ForcedNextRoomSide.Value;
-          else
-          {
-            if (currentAppendInfo.side == EntranceSide.Bottom)
-              nextAppendInfo.side = EntranceSide.Right;
-            else if (currentAppendInfo.side == EntranceSide.Right)
-              nextAppendInfo.side = EntranceSide.Bottom;
-          }
+        if ((i % 2) == 0){
+          heightOfPassage = ((mazeNodes[h].Height / 2) - 1) + currentHeight;
+          widthOfPassage = mazeNodes[h].Width - 1;
+          h++;
+          currentHeight = distances[0];
         }
-        if (lastNonSecretNode != null)
-        {
-          sizeProvider = lastNonSecretNode;
-          nextAppendInfo.Position = appendNodeInfos[lastNonSecretNode].Position;
+        if ((i % 2) == 1){ //h
+          heightOfPassage = mazeNodes[h1].Height - 1;
+          widthOfPassage = mazeNodes[h1].Width / 2 + currentWidth - 2;
+          currentWidth = distances[1];
+          h1 += 2;
         }
-      }
-      else
-      {
-        //if (currentNodeIndex == 0)
-        //  nextAppendInfo.side = EntranceSide.Bottom;
-        //else
-        nextAppendInfo.side = CalcSide(currentAppendInfo.side, nextAppendInfo.side, chanceForLevelTurn);
+          level.AppendMaze
+        (
+          mazeNodes[i + 5],
+          new Point(widthOfPassage, heightOfPassage)
+        );
       }
 
-      var pt = nextAppendInfo.Position;
-      if (nextAppendInfo.side == EntranceSide.Bottom)
-      {
-        pt.Y += sizeProvider.Height - 1 + nodesPadding; //SK -1 => +2
-      }
-      else if (nextAppendInfo.side == EntranceSide.Right)
-      {
-        pt.X += sizeProvider.Width - 1;//SK - 1 => +2
-      }
-      nextAppendInfo.Position = pt;
+      heightOfPassage = (mazeNodes[0].Height / 2) + 1;
+      widthOfPassage = 12 + (mazeNodes[4].Width / 2) - 1;
```

Comment:
Wiem że to wczesna wersja, ale nigdy nie używaj "magicznych" liczb, tylko deklaruj nazwane zmienne, stałe.
Oczywiście nie mam na mysli konstrukcji typu : if ((i % 2) - tu jest ok, tylko inne przypadki gdzie robimy jakąś logikę i można użyć stałej/zmiennej.